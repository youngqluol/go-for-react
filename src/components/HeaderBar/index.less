.header-bar {
  width: 100px;
  height: 100px;
  .text-color {
    color: blue;
  }
}
