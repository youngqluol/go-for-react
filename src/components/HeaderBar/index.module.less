.headerBar {
  width: 100px;
  height: 100px;
  .textColor {
    color: blue;
  }
}
