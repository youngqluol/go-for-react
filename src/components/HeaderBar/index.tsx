import { Button } from 'antd';
import { observer } from 'mobx-react';
import HeaderStyle from './index.module.less';
import { useStores } from '../../store';

interface HeaderBarProps {
  text: string;
}

function HeaderBar(props: HeaderBarProps) {
  const { counterStore } = useStores();
  const { counter, increment } = counterStore;
  return (
    <div className={HeaderStyle.headerBar}>
      <p className={HeaderStyle.textColor}>{props.text}</p>
      <Button type='primary'>Primary Button</Button>
      <h2>headBar counter:{counter}</h2>
      <button type='submit' onClick={e => increment()}>
        +1
      </button>
    </div>
  );
}

export default observer(HeaderBar);
