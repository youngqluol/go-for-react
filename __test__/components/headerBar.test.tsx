import React from 'react';
import renderer from 'react-test-renderer';
import HeaderBar from '../../src/components/HeaderBar';

// describe 描述一组测试
describe('test HeaderBar component render', () => {
  it('test Movebox component render success', () => {
    // renderer.create 模拟 React 渲染组件，并将结果输出为 json 字符串。
    const moveComp = renderer.create(<HeaderBar text='text' />).toJSON();
    // 期望 HeaderBar 匹配生成的快照文件。
    expect(moveComp).toMatchSnapshot();
  });
});
