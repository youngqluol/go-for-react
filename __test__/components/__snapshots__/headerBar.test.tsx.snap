// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test HeaderBar component render test Movebox component render success 1`] = `
<div
  className="header-bar"
>
  <p
    className="text-color"
  >
    text
  </p>
</div>
`;
